demo-app/src/bank_cmd/main.rs

```diff
@@ -0,0 +1,317 @@
+use anyhow::Context;
+use borsh::BorshSerialize;
+use clap::Parser;
+use demo_app::{runtime::Runtime, sign_tx, Transaction};
+use sov_app_template::RawTx;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+use sov_modules_api::{
+    default_context::DefaultContext, default_signature::private_key::DefaultPrivateKey, PublicKey,
+    Spec,
+};
+
+type C = DefaultContext;
+type Address = <C as Spec>::Address;
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+struct PrivKeyAndAddress {
+    hex_priv_key: String,
+    address: Address,
+}
+
+impl PrivKeyAndAddress {
+    fn generate() -> Self {
+        let priv_key = DefaultPrivateKey::generate();
+        let address = priv_key.pub_key().to_address();
+        Self {
+            hex_priv_key: priv_key.as_hex(),
+            address,
+        }
+    }
+
+    fn generate_and_save_to_file(priv_key_path: &Path) -> anyhow::Result<()> {
+        let priv_key = Self::generate();
+        let data = serde_json::to_string(&priv_key)?;
+        fs::create_dir_all(priv_key_path)?;
+        let path = Path::new(priv_key_path).join(format!("{}.json", priv_key.address));
+        fs::write(path, data)?;
+        Ok(())
+    }
+}
+
+#[derive(Parser)]
+enum Cli {
+    /// Creates a new private key.
+    CreatePrivateKey {
+        /// Location of the private key.
+        priv_key_path: String,
+    },
+    /// Serializes call message.
+    SerializeCall {
+        /// Private key used to sign the transaction.
+        sender_priv_key_path: String,
+        /// Location of the `call message`.
+        call_data_path: String,
+        /// The `call message` nonce.
+        nonce: u64,
+    },
+}
+
+fn main() {
+    let cli = Cli::parse();
+
+    match cli {
+        Cli::CreatePrivateKey { priv_key_path } => {
+            PrivKeyAndAddress::generate_and_save_to_file(priv_key_path.as_ref())
+                .unwrap_or_else(|e| panic!("Create private key error: {}", e));
+        }
+        Cli::SerializeCall {
+            sender_priv_key_path,
+            call_data_path,
+            nonce,
+        } => {
+            let serialized = SerializedTx::new(&sender_priv_key_path, &call_data_path, nonce)
+                .unwrap_or_else(|e| panic!("Call message serialization error: {}", e));
+
+            let mut bin_path = PathBuf::from(call_data_path);
+            bin_path.set_extension("bin");
```

Comment:
nit: The `.bin` file extension makes me think this is an (executable) binary. Would it be clearer if we used `.dat` instead?

Reply:
done